### PR TITLE
[Examples] Add attribute list example

### PR DESCRIPTION
**Pull request by ClydeZhao**

### CODE REVIEW

**Comment by ClydeZhao** [comment]
@ddmbr @Yuzhen11 @kygx-legend Please review.😀

File: examples/pr_cc_attrlist.cpp [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    friend husky::BinStream& operator<<(husky::BinStream& stream, const Vertex& v) {
+        stream << v.id_ << v.neighbors_;
+        return stream;
+    }
+    friend husky::BinStream& operator>>(husky::BinStream& stream, Vertex& v) {
+        stream >> v.id_ >> v.neighbors_;
+        return stream;
+    }
+
+    KeyT id_;
```
Comment (kygx-legend) [inline examples/pr_cc_attrlist.cpp]: Don't need to name public members with `_` in the end.

File: examples/connected_components.cpp [hunk 2]
```diff
@@ -26,22 +26,22 @@ class Vertex {
...
+        stream << u.vertexId << u.adj << u.cid;
         return stream;
     }
     friend husky::BinStream& operator>>(husky::BinStream& stream, Vertex& u) {
-        stream >> u.vertexId >> u.adj >> u.color;
+        stream >> u.vertexId >> u.adj >> u.cid;
         return stream;
     }
 
     int vertexId;
```
Comment (kygx-legend) [inline examples/connected_components.cpp]: Could you use `vertex_id`?

File: examples/pr_cc_attrlist.cpp [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    friend husky::BinStream& operator<<(husky::BinStream& stream, const Vertex& v) {
+        stream << v.VertexId << v.neighbor;
+        return stream;
+    }
+    friend husky::BinStream& operator>>(husky::BinStream& stream, Vertex& v) {
+        stream >> v.VertexId >> v.neighbor;
+        return stream;
+    }
+
+    KeyT VertexId;
```
Comment (kygx-legend) [inline examples/pr_cc_attrlist.cpp]: Same. `vertex_id`.

**Pull request merged**